### PR TITLE
fix: grafana view only permission

### PR DESCRIPTION
**Pull request by cwyl02**
**What this PR does/ why we need it**:
requires the * after /proxy due to https://grafana.com/docs/grafana/latest/http_api/data_source/#data-source-proxy-calls

https://jira.d2iq.com/browse/D2IQ-88865

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: staging/kube-prometheus-stack/templates/ingress-rbac/grafana.yaml [hunk 1]
```diff
@@ -36,11 +36,12 @@ rules:
   - nonResourceURLs:
     - {{ .Values.grafana.ingress.path | trimSuffix "/" }}
     - {{ .Values.grafana.ingress.path | trimSuffix "/" }}/*
+    - {{ .Values.grafana.ingress.path | trimSuffix "/" }}/api/datasources/proxy/*
     verbs:
       - get
       - head
   - nonResourceURLs:
-    - {{ .Values.grafana.ingress.path | trimSuffix "/" }}/api/datasources/proxy/1/api/v1/series
+    - {{ .Values.grafana.ingress.path | trimSuffix "/" }}/api/datasources/proxy/*
```
Comment (cbuto) [inline staging/kube-prometheus-stack/templates/ingress-rbac/grafana.yaml:44]: do we still need to allow this route as a POST? I don't see anything mentioning it in the docs 🤔

Comment (cwyl02) [reply staging/kube-prometheus-stack/templates/ingress-rbac/grafana.yaml:44]: Yes, the POST is a must -- its not on Grafana Doc but I see the error from Chrome DevTool. I guess that's an internal API

Comment (cwyl02) [reply staging/kube-prometheus-stack/templates/ingress-rbac/grafana.yaml:44]: saw errors /api/datasources/proxy/1/api/v1/series and /api/datasources/proxy/2/api/v1/series (when multiple data sources are available)

**Review by gracedo — COMMENTED** [review]
Can you apply the same change in the patch template here: https://github.com/mesosphere/charts/blob/master/staging/kube-prometheus-stack/patch/mesosphere/templates/ingress-rbac/grafana.yaml

This chart utilizes patch scripts to apply mesosphere-specific patches on top of the upstream chart, if we don't include the change in the patch then this might get overridden the next time someone runs the patch scripts

File: staging/kube-prometheus-stack/templates/ingress-rbac/grafana.yaml (hunk 1 above)
Comment (cbuto) [reply staging/kube-prometheus-stack/templates/ingress-rbac/grafana.yaml:44]: ah gotcha, thanks!

**Review by makkes — APPROVED** [review]

**Comment by makkes** [comment]
```
   ✖︎ kube-prometheus-stack => (version: "34.9.0", path: "staging/kube-prometheus-stack") > Chart version not ok. Needs a version bump!
```

**Review by cbuto — APPROVED** [review]

**Review by gracedo — APPROVED** [review]
Can you apply the same change in the patch template here: https://github.com/mesosphere/charts/blob/master/staging/kube-prometheus-stack/patch/mesosphere/templates/ingress-rbac/grafana.yaml

This chart utilizes patch scripts to apply mesosphere-specific patches on top of the upstream chart, if we don't include the change in the patch then this might get overridden the next time someone runs the patch scripts

**Comment by gracedo** [comment]
that's weird, it reposted my previous comment lol. anyways, merging

**Pull request merged**